PennMobile/GSR-Booking/Controllers/GSRController.swift

```diff
@@ -146,12 +157,16 @@ extension GSRController: GSRViewModelDelegate {
                     self.stopLoadingViewAnimation()
                 }
             }
+            
         }
+        
+        
     }
 
     func refreshDataUI() {
-        emptyView.isHidden = !viewModel.isEmpty
         tableView.isHidden = viewModel.isEmpty
+        closedView.isHidden = viewModel.existsNonEmptyRoom()
```

Comment:
Ahh .. I totally see why `existsNonEmptyRoom` is not good enough for our purposes here. One way to resolve it:

Instead of checking whether at least one `room` exists, we should check if at least one `timeSlot` exists! If 0 `timeSlot`'s exists, then we should show the `closedView`. Otherwise, we should hide it.

Reply:
Just want to add something here -- I think we may want to add a different empty state rather than "closed" for when there are no empty rooms available. Def agree with Rehaan here, "Closed" should be used for an empty schedule for the building, which would signify that it's being closed for a significant amount of time"